services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,944 @@
+        def protectedIslsInfo = protectedIsls.collect { islUtils.getIslInfo(it).get() }
+
+        allIsls.each { isl ->
+            protectedIslsInfo.each { protectedIsl ->
+                if (isl.id == protectedIsl.id) {
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        when: "Init reroute"
+        def islToBreak = pathHelper.getInvolvedIsls(currentPath)[0]
+        northbound.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+
+        then: "Flow is rerouted"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow.id).status == FlowState.UP
+            assert pathHelper.convert(northbound.getFlowPath(flow.id)) != currentPath
+        }
+
+        and: "Current path is protected path"
+        def flowPathInfoAfterRerouting = northbound.getFlowPath(flow.id)
+        pathHelper.convert(flowPathInfoAfterRerouting) == currentProtectedPath
+        pathHelper.convert(flowPathInfoAfterRerouting.protectedPath) == currentPath
+
+        when: "Restore port status"
+        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Path of the flow is not changed"
+        pathHelper.convert(northbound.getFlowPath(flow.id)) == currentProtectedPath
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+
+        where:
+        flowDescription | bandwidth
+        "metered"       | 1000
+        "unmetered"     | 0
+    }
+
+    @Unroll
+    def "System reroutes #flowDescription flow to more preferable path and ignore protected path in case reroute is intentional"() {
+        given: "A flow with alternate paths available"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = database.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 2
+        } ?: assumeTrue("No suiting switches found", false)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        currentPath != currentProtectedPath
+
+        when: "Make the current and protected path less preferable than alternatives"
```

Comment:
I think making the current and protected paths less preferable than alternatives is incorrect. You should make the protected path the most preferable path, alternative paths should be more preferable than the current one, but less preferable than protected.

Reply:
The goal of the test is to verify that the best path will be picked in case of manual reroute, opposite to the case when protected path is forced during auto-reroute (even if better path are available). If we change to your option, then we will result in flow rerouting to protected path (which is the same as during auto-reroute). What will this test test in this case?

Reply:
I thought that we should not reroute to the protected path when we do a manual reroute even though the protected path is the most preferable. And I suggested verifying this.

Reply:
leave it as is